src/Icicle/Avalanche/Prim/Eval.hs

```diff
@@ -111,7 +111,7 @@ evalPrim p vs
       -> primError
 
 
-     PrimUpdate (PrimUpdateArrayPut t)
+     PrimArray (PrimArrayPutMutable t)
```

Comment:
hmm after digging around in the tests, I think making this evaluator do actual mutation is the easiest way to test the `mutate` transform